### DISCUSSION TITLE
Tips for coming from KDE?

### DISCUSSION THREAD

**Discussion by digitalsignalperson** [WayfireWM/wayfire#2217] [General]
I've poked at Wayfire a few times but have yet to try daily driving it. Since putting so many hours into Plasma 6 bug reports recently, I can't help but wonder what if I put those hours into figuring out Wayfire. Maybe this is a good time and opportunity to market wayfire as an alternative (maybe dotfiles for post-kde starter pack?). It seems like the best floating-window alternative, where hyprland and sway seem to be the most popular alternatives to kde/gome. 

What are the biggest gaps or things to get used to?

My favorite things in KDE:
- vertical panel (but I'm seemingly a rare user of. it has regressions. and getting the vibe it could just get deprecated like Window 11 did). I remember looking around before and this was something hard to find in the bespoke wayland panel land. And hard to find panels that displayed both window icons and their titles. I wouldn't mind making this from scratch if necessary, if the data is easily avail over IPC. (probs https://github.com/WayfireWM/wayfire/wiki/Tips-&-Tricks#getting-the-title-and-app-id-of-windows)
- everything contained in system settings. I did take a poke at wf-shell before, which probably provides a subset of the same
- kwin scripting (wayfire IPC stuff looks good)
- desktop effects (obviously wayfire great here)
- built in graphical display configuration (kde now has HDR and easy to config refresh rates, adaptive sync, etc.)
- everything being in the official arch linux repos is a draw to me. 

Per the last point, I have trust issues and have a hard time wanting to maintain reviewing and buidling many AUR-dependencies to piece together a desktop environment.  Any outlook on how much more adoption wayfire needs to get that level of acceptance? Hyperland managed to make it in the last year or so. 

Stuff that is annoying in KDE:
- the mix of dbus, qml, C++. everything is so all over the place and hard to trace.
- bloat. lots of hacks to disable things like baloo, discover,  kwallet, and then dealing with associated bugs
- multi-monitor stuff still buggy
- qml is favored. I keep trying to maintain my own C++ kwin effects, but it's a battle to keep it running. And I don't even want to try porting to Qt 6 with no support or examples.
- lacking various protocols that are constantly added to wlroots
- poor support for touch and gestures

I'm also checking out stuff in the wiki. I may try some dotfiles [from peux-os](https://github.com/peux-os-sic/Peux-wayfire-22). PaperDE looks cool too. Also maybe CachyOS https://github.com/CachyOS/cachyos-wayfire-settings

**Comment by ammen99** [comment]
Hi. Your post contains quite a lot of questions, I'll try to answer as best as I can ;)

> * vertical panel (but I'm seemingly a rare user of. it has regressions. and getting the vibe it could just get deprecated like Window 11 did). I remember looking around before and this was something hard to find in the bespoke wayland panel land. And hard to find panels that displayed both window icons and their titles. I wouldn't mind making this from scratch if necessary, if the data is easily avail over IPC. (probs https://github.com/WayfireWM/wayfire/wiki/Tips-&-Tricks#getting-the-title-and-app-id-of-windows)

Our panel, wf-panel, does not support vertical layout yet. However, there might be other panels/bars that support this. Getting the title and app-id is already possible by using wayland protocol extensions which have been standardized across wlroots compositors, or you can use the IPC - this would be tied to Wayfire specifically, but you get even more possibilities in what to do with the windows. Maybe take a look at what @killown is doing - https://github.com/killown/waypanel .

> * everything contained in system settings. I did take a poke at wf-shell before, which probably provides a subset of the same

Wayfire is not a full DE, as such, many of those settings will be missing and they are not even planned for inclusion. But, you should be able to take KDE's Settings app, start the necessary services (upower, etc.) and many things will work. I regularly use gnome-control-center for example.

> * built in graphical display configuration (kde now has HDR and easy to config refresh rates, adaptive sync, etc.)

In Wayfire's config file, you can specify VRR, refresh rates, modes, scale, position, rotation. HDR is not supported - I am not really familiar with color management, and am generally waiting until all Wayland bits are finished, so that it can be implemented together.

> * qml is favored. I keep trying to maintain my own C++ kwin effects, but it's a battle to keep it running. And I don't even want to try porting to Qt 6 with no support or examples.

Wayfire only supports C++, but if you're doing some simpler scripts, you can use the IPC which has a simple format and can be used in pretty much any language which supports unix sockets and json.

> * poor support for touch and gestures

Wayfire does have some support for these. Whether they would be enough for you or not, I cannot say.

> I'm also checking out stuff in the wiki. I may try some dotfiles [from peux-os](https://github.com/peux-os-sic/Peux-wayfire-22). PaperDE looks cool too. Also maybe CachyOS https://github.com/CachyOS/cachyos-wayfire-settings

Yeah, maybe one thing which needs to be said about Wayfire for someone coming from KDE is this: Wayfire is more similar to kwin. But, to get a fully functional desktop as you want it, you need the full 'Plasma' setup. This includes panels and backgrounds (provided by wf-shell or you can use custom components, like swaybg, waybar, etc.), but also a lot of integration with other services which is not present in Wayfire. Some of those projects you mention are trying to do that - integrate Wayfire better with the rest of the system so that you get a working desktop out of the box.

> Per the last point, I have trust issues and have a hard time wanting to maintain reviewing and buidling many AUR-dependencies to piece together a desktop environment. Any outlook on how much more adoption wayfire needs to get that level of acceptance?

Unfortunately, no idea about that.

**Reply by digitalsignalperson** [reply]
Thanks for the thoughts. I look forward to exploring this some more.

**Comment by NOSHIRTEDKEEPER** [comment]
Is there a list somewhere of other app options to install to get the full setup?
